airflow/cli/commands/connection_command.py

```diff
@@ -140,20 +136,19 @@ def connections_export(args):
             _, filetype = os.path.splitext(args.file.name)
             filetype = filetype.lower()
             if filetype not in allowed_formats:
-                msg = (
-                    f"Unsupported file format. "
-                    f"The file must have the extension {', '.join(allowed_formats)}"
+                raise SystemExit(
+                    f"Unsupported file format. The file must have "
+                    f"the extension {', '.join(allowed_formats)}."
                 )
-                raise SystemExit(msg)
 
         connections = session.query(Connection).order_by(Connection.conn_id).all()
         msg = _format_connections(connections, filetype)
         args.file.write(msg)
 
         if _is_stdout(args.file):
-            print("Connections successfully exported.", file=sys.stderr)
+            print(f"Connections successfully exported to {args.file.name}.")
         else:
-            print(f"Connections successfully exported to {args.file.name}")
+            print("Connections successfully exported.", file=sys.stderr)
```

Comment:
Can you explain this change please

Reply:
I think this changes itroduces expected behaviour. If the file is `stdout` then print to `stdout` ot `stderr`. But I can be wrong, I think this was done by @mik-laj

Reply:
If we display the output on stdout, we cannot write another message to that stream.

Reply:
Before:
```
$ airflow connections export - | jq .
....
  "yandexcloud_default": {
    "conn_type": "yandexcloud",
    "description": null,
    "host": null,
    "login": null,
    "password": null,
    "schema": "default",
    "port": null,
    "extra": null
  }
}
```
After
```
$ airflow connections export - | jq .
.....
    "login": null,
    "password": null,
    "schema": "default",
    "port": null,
    "extra": null
  }
}
parse error: Invalid numeric literal at line 472, column 13
```

Reply:
It works fine on the master branch

Reply:
We should probably unify this method to work with the `output` argument as in #12704, this will introduce more consistency. I will do this in next PR